lib/src/models/message/component.dart

```diff
@@ -22,10 +37,76 @@ final class MessageComponentType extends EnumLike<int, MessageComponentType> {
   MessageComponentType.parse(int value) : this(value);
 }
 
+class UnfurledMediaItem with ToStringHelper implements CdnAsset {
+  /// The manager for this [UnfurledMediaItem].
+  final MessageManager manager;
+
+  @override
+  Nyxx get client => manager.client;
+
+  /// The URL of this media item.
+  @override
+  final Uri url;
+
+  /// A proxied URL of this media item.
+  final Uri? proxiedUrl;
+
+  /// The height of this media item if it is an image.
+  final int? height;
+
+  /// The width of this media item if it is an image.
+  final int? width;
+
+  /// @nodoc
+  UnfurledMediaItem({
+    required this.manager,
+    required this.url,
+    required this.proxiedUrl,
+    required this.height,
+    required this.width,
+  });
+
+  @override
+  HttpRoute get base => throw UnsupportedError('Cannot get the base URL for a media item');
+
+  @override
+  CdnFormat get defaultFormat => throw UnsupportedError('Cannot get the default format for a media item');
+
+  @override
+  String get hash => throw UnsupportedError('Cannot get the hash for a media item');
+
+  @override
+  bool get isAnimated => false;
+
+  @override
+  Future<Uint8List> fetch({CdnFormat? format, int? size}) async {
+    if (format != null || size != null) {
+      throw UnsupportedError('Cannot specify attachment format or size');
+    }
+
+    final response = await client.httpHandler.httpClient.get(url);
+    return response.bodyBytes;
+  }
+
+  @override
+  Stream<List<int>> fetchStreamed({CdnFormat? format, int? size}) async* {
+    if (format != null || size != null) {
+      throw UnsupportedError('Cannot specify attachment format or size');
+    }
+
+    final response = await client.httpHandler.httpClient.send(Request('GET', url));
+    yield* response.stream;
+  }
+}
+
 /// A component in a [Message].
 abstract class MessageComponent with ToStringHelper {
   /// The type of this component.
   MessageComponentType get type;
+
+  final int id;
+
+  MessageComponent({required this.id});
```

Comment:
Isnt this breaking?

(Well, technically, the odds that someone actually extends `MessageComponent` is very low but still)

Reply:
fair